clash-lib/src/Clash/Backend/SystemVerilog.hs

```diff
@@ -1126,7 +1126,7 @@ expr_ _ (Identifier id_ (Just (Indexed ((Unsigned w),_,_))))  = do
 expr_ _ (Identifier _ (Just (Indexed ((BitVector _),_,0)))) = do
   iw <- Ap $ use intWidth
   traceIf True ($(curLoc) ++ "WARNING: synthesizing bitvector mask to dontcare") $
-    verilogTypeErrValue (Signed iw)
+    verilogTypeErrValue (Unsigned iw)
```

Comment:
The commit message reads:

> BitVector masks were changed to use Natural or unsigned, but these
codepaths weren't changed.

I'm not sure what this means. What PR introduced this?

Reply:
505604a33

Reply:
The _or unsigned_ is really unclear to me. Should it have read

> ... to use `Natural` for `Unsigned` and `BitVector`

?

Reply:
I don't think our data type `Unsigned` has anything to do with it though? I did fix a few typo's in this commit message, so it might be another one. I also thought it was curiously worded, but I interpreted "Natural or unsigned" as "Natural, which is unsigned rather than signed". I'll shoot Leon a message what he meant.